runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/StreamingViewOverrides.java

```diff
@@ -63,8 +61,21 @@ private StreamingCreatePCollectionView(PCollectionView<ViewT> view) {
 
       @Override
       public PCollection<ElemT> expand(PCollection<ElemT> input) {
-        return input
-            .apply(Combine.globally(new Concatenate<ElemT>()).withoutDefaults())
+        PCollection<List<ElemT>> elements;
+        if (view.getViewFn() instanceof PCollectionViews.IsSingletonView) {
```

Comment:
you can use the URN of the ViewFn I believe

Reply:
Does that work on runner v1 as well?

Reply:
This code path is only active on v1, and yea the URNs were put in place for that purpose. BTW this particular change would only be for the case where the upstream global combine triggers twice before the concat triggers once, right?

Reply:
the URN of a singleton view is beam:side_input:iterable:v1 unfortunately, so we can't use that.

Reply:
The view's materialization should be iterable, but the ViewFn itself is responsible for converting the iterable to a singleton. In v1 there shouldn't actually be a separate materialization concept, and ViewFn is an implementation detail, not part of the model. So maybe things are a bit tangled. I'm OK with the change here, but ideally we wouldn't introduce another class to the public API. Is the new class fully `@Internal`?

Reply:
Yes, it's marked Internal